[PATCH v4] API: add new crypto algorithms

JannePeltonen opened the pull request:
Add new crypto algorithms in the API. This PR does not add real validation tests for the algorithms but just makes the validation test aware of the new algorithms.

psavol commented on test/validation/api/crypto/util.c:
```diff
@@ -102,6 +124,20 @@ const char *cipher_alg_name(odp_cipher_alg_t cipher)
+	case ODP_CIPHER_ALG_SM4_ECB:
+		return "ODP_CIPHER_ALG_SM4_ECB";
+	case ODP_CIPHER_ALG_SM4_CBC:
+		return "ODP_CIPHER_ALG_CM4_CBC";
```
Typo: _CM4_CBC -> _SM4_CBC

JannePeltonen commented:
v2:
- fixed typos
- improved SNOW-V-GMAC and SM4-GMAC documentation slightly
- improved auth_range documentation to refer to GMAC algorithms in general
- rebased

anoobj commented:
The change looks good to me.

Acked-by: Anoob Joseph <anoobj@marvell.com>

anoobj commented:
The change looks good to me.

Acked-by: Anoob Joseph <anoobj@marvell.com>

JannePeltonen commented:
v3: added reviewed-by and acked-by tags
